--- a/src/libcamera/meson.build
+++ b/src/libcamera/meson.build
@@ -70,7 +70,7 @@
 libatomic = cc.find_library('atomic', required : false)
 libthreads = dependency('threads')
 
-mesa_works = cc.check_header('EGL/egl.h', required: false)
+mesa_works = cc.check_header('EGL/egl.h', required: false) and cc.check_header('GLES2/gl2.h', required: false)
 libegl = dependency('egl', required : false)
 libglesv2 = dependency('glesv2', required : false)
 
